ee/clickhouse/queries/cohort_query.py

```diff
@@ -0,0 +1,476 @@
+from typing import Any, Dict, List, Optional, Set, Tuple, Union
+
+from ee.clickhouse.materialized_columns.columns import ColumnName
+from ee.clickhouse.models.cohort import get_count_operator, get_entity_query
+from ee.clickhouse.models.property import prop_filter_json_extract
+from ee.clickhouse.queries.event_query import EnterpriseEventQuery
+from posthog.models import Filter, Team
+from posthog.models.action import Action
+from posthog.models.filters.mixins.utils import cached_property
+from posthog.models.filters.path_filter import PathFilter
+from posthog.models.filters.retention_filter import RetentionFilter
+from posthog.models.filters.session_recordings_filter import SessionRecordingsFilter
+from posthog.models.filters.stickiness_filter import StickinessFilter
+from posthog.models.property import Property, PropertyGroup, PropertyName
+from posthog.queries.column_optimizer import ColumnOptimizer
+
+Relative_Date = Tuple[int, str]
+NOW: Relative_Date = (0, "now")
+
+Period = Tuple[Relative_Date, Relative_Date]
+Event = Tuple[str, Tuple[str, int]]
+Event_In_Period = Tuple[Event, Period]
+
+USES_PERIOD = ["performed_event", "performed_event_muliple", "stopped_performing_event", "restarted_performing_event"]
+
+
+class CohortOptimizer(ColumnOptimizer):
```

Comment:
Q: Why is this related to `ColumnOptimizer`?

It seems like its mostly getting the column names to query, but also these column names are ones we're creating ourselves specifically for cohort filtering only